drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -11,36 +11,48 @@
 #include <drivers/uart.h>
 #include <kernel.h>
 #include <pm/device.h>
+#include <pm/pm.h>
+#include <pm/policy.h>
 #include <soc.h>
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(uart_ite_it8xxx2, LOG_LEVEL_ERR);
 
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+static struct uart_it8xxx2_data *uart_data;
+#endif
+
 struct uart_it8xxx2_config {
 	uint8_t port;
 	/* GPIO cells */
 	struct gpio_dt_spec gpio_wui;
 };
 
+struct uart_it8xxx2_data {
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+	struct k_work_delayable rx_refresh_timeout_work;
+#endif
+};
+
 enum uart_port_num {
 	UART1 = 1,
 	UART2,
 };
 
 #ifdef CONFIG_PM_DEVICE
-__weak void uart1_wui_isr_late(void) { }
-__weak void uart2_wui_isr_late(void) { }
-
 void uart1_wui_isr(const struct device *gpio, struct gpio_callback *cb,
 		   uint32_t pins)
 {
 	/* Disable interrupts on UART1 RX pin to avoid repeated interrupts. */
 	(void)gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
 					   GPIO_INT_DISABLE);
 
-	if (uart1_wui_isr_late) {
-		uart1_wui_isr_late();
-	}
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+	k_timeout_t delay = K_MSEC(CONFIG_UART_CONSOLE_INPUT_EXPIRED_TIMEOUT);
+
+	pm_constraint_set(PM_STATE_STANDBY);
+	k_work_reschedule(&uart_data->rx_refresh_timeout_work, delay);
+#endif
```

Comment:
nit: Can you add a comment for readers as to why you are going standby and why it is only needed for UART1?

Reply:
Added comment. 
And added to judge whether uart is a console device to use console input expired.